api/python/provisioner/commands/check.py

```diff
@@ -220,13 +266,55 @@ def get_passed(self) -> list:
         """
         return [check for check in self._check_entries if check.is_passed]
 
-    def get_failed(self) -> list:
+    def get_failed(self, critical: bool = None) -> list:
```

Comment:
we don't follow typing strictly but for the sake of accuracy and just if you are not aware: `Optional[bool]` I believe would be the best choice here

Reply:
Yes, I also think it would be better here and for `get_checks` method too.